src/interfaces/wallet.cpp

```diff
@@ -65,7 +65,7 @@ WalletTx MakeWalletTx(CWallet& wallet, const CWalletTx& wtx)
     for (const auto& txin : wtx.tx->vin) {
         result.txin_is_mine.emplace_back(wallet.IsMine(txin));
         // NOTE: We check only inputs our wallet knows about, cause IsDenominated returns "false" for non-wallet txes
-        // and finding/fetching the actual tx from blocks db locks cs_main which we should (and can) avoid here.
+        // and finding/fetching the actual tx from mapWallet locks cs_wallet which we should (and can) avoid here.
```

Comment:
No, I mean finding that tx requires scanning blocks db via GetTransaction which locks cs_main

Reply:
Hm i guess i can't really follow your thoughts here right now, can you explain how/where exactly 004752b avoids calling `GetTransaction`?

Reply:
I mean `wallet.IsDenominated(...)` works only for inputs that are known to _our_ wallet and many inputs in mixing tx do not belong to it. To check all inputs we can do f4e4973aef (and then aaea64b170 some parts of which are required anyway ;) ). But I just realized that `cs_main` is locked whenever we call `MakeWalletTx` and this lock inside `GetTransaction` is not an issue really, accessing tx index is probably fine too 🤔

Reply:
@UdjinM6 Can you check ac3a8a0? I thiiiink that should be good enough?